src/ClientMessage.ts

```diff
@@ -224,7 +224,7 @@ class ClientMessage {
     readBuffer(): Buffer {
```

Comment:
This method is used for reading Data in `readData` method. Not all fields of Data is set to default values after that. Garbage values can lead to wrong behavior. Check especially HeapData.getPartitionHash method.

Reply:
The `ClientMessage#readBuffer` method is not affected by this PR at all. And I don't see how it's related with proposed changes. Are you describing a separate issue? Could you clarify?

Reply:
`const result = new Buffer(size);` is converted to `Buffer.allocUnsafe(size);` in `readBuffer`. 
I thought that it is allocated here and will be filled later. Now, I see that this is the part that we directly copy from internal `buffer` to `result`. So ignore my comment. I don't see a problem here at the moment.